### PR TITLE
GWTP-121 - presenters documentation

### PR DESCRIPTION
**Pull request by Pacane**

### CODE REVIEW

File: site/src/main/markdown/gwtp_v2/core/presenters/creating-places.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Creating places: Presenter Proxy
+A *Place* represents a particular 'bookmark' or location inside the application. A Place is stateful - it may represent a location with it's current settings, such as a particular ID value, or other unique indicator that will allow a user to track back to that location later, either via a browser bookmark, or by clicking the 'back' button.
```
Comment (olafleur) [inline site/src/main/markdown/gwtp_v2/core/presenters/creating-places.md]: its current settings (pas d'apostrophe)

File: site/src/main/markdown/gwtp_v2/core/presenters/gin-bindings.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+GWTP uses dependency injection as a foundation of its architecture. It allows to have a maintainable application and loosely coupled components.
+
+GIN is the library used for dependency injection in GWT.
+
+## References
+[Google GIN](http://code.google.com/p/google-gin/)
+
+## Binding MVP
+GWTP needs a boostrapping process to link a Presenter class to it's View and View implementation class. The way of doing this is by creating a GIN module and do the setup the binding of classes and interfaces in there.
```
Comment (olafleur) [inline site/src/main/markdown/gwtp_v2/core/presenters/gin-bindings.md]: its

Comment (olafleur) [inline site/src/main/markdown/gwtp_v2/core/presenters/gin-bindings.md]: classes

Comment (olafleur) [inline site/src/main/markdown/gwtp_v2/core/presenters/gin-bindings.md]: aren't you missing a comma somewhere at the end of this sentence.

File: site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Lifecycle
+
+A PresenterWidget has multiple phases in its lifecycle. GWTP gives the user access to many handles to execute code during each phases.
+
+Let's see what the phases are, and then describe each of them with a possible use case for each on of them.
+
+<!--- show lifecycle diagram -->
+![Diagram](http://i.imgur.com/kwjjGuv.png)
```
Comment (olafleur) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: I think there is a prettier graphic in a presentation that @christiangoudreau gave recently.

**Comment by olafleur** [comment]
LGTM

File: site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md (hunk 3 above)
Comment (Pacane) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: got the image file somewhere?

Comment (Pacane) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: @jasonlemay how can I add an image in there? halp me css ninja.

File: site/src/main/markdown/gwtp_v2/core/presenters/creating-places.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## Proxy
+Proxies are light-weight classes whose size doesn't depend on the complexity of the underlying Presenter and View. They are instantiated as soon as the application loads and are responsible for listening to any event that would require their associated Presenter and View to be created. Proxies are the key to a fast MVP web application, they enable code splitting and lazy instantiation of the largest part of your code.
+
+Chances are you will never have to write a Proxy class. That's because most of the time Proxies can be created automatically by GWT's generators. All you have to do is use special annotations above your `MyProxy` interface.
+
+## CodeSplit
+The first annotation you will need is either `@ProxyStandard` or `@ProxyCodeSplit`, depending on whether or not you want your Presenter and View to sit behind a split point (see [here](http://code.google.com/webtoolkit/doc/latest/DevGuideCodeSplitting.html) for details on GWT's code splitting feature).
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/creating-places.md]: Maybe `@ProxyStandard`,  `@ProxyCodeSplit` or  `@ProxyCodeSplitBundle`?

File: site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Lifecycle
+
+A PresenterWidget has multiple phases in its lifecycle. GWTP gives the user access to many handles to execute code during each phases.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: phase

File: site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+Let's see what the phases are, and then describe each of them with a possible use case for each on of them.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: each one

File: site/src/main/markdown/gwtp_v2/core/presenters/popups.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
+# Popup Presenter
+A popup presenter is a standard presenter with a view that extends PopupView.
+
+##Reference
+* [Popup Presenter Example](https://github.com/ArcBees/GWTP-Samples/tree/master/gwtp-samples/gwtp-sample-tab/src/main/java/com/gwtplatform/samples/tab/client/application/infopopup)
+* [Add Popup Presenter](https://github.com/ArcBees/GWTP-Samples/blob/master/gwtp-samples/gwtp-sample-tab/src/main/java/com/gwtplatform/samples/tab/client/application/globaldialog/GlobalDialogSubTabPresenter.java#L79)
+* [Popup slot]({{#gwtp.doc.url.slots}})
+
+##Display a Popup Presenter
+The Popup Presenter skips the view's slot methods all together. When it is used it adds directly to the DOM and requests displays in the center or location selected.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: altogether

Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: Maybe change second sentence to : `When it is used, it is added directly to the DOM and displayed at the specified location (centered by default).`

File: site/src/main/markdown/gwtp_v2/core/presenters/popups.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    @Inject
+    InfoPopupPresenterWidget(EventBus eventBus,
+                             MyView view) {
+        super(eventBus, view);
+    }
+}
+```
+
+##View
+The view extends `PopupViewImpl` or `PopupViewWithUiHandlers<PresentersUiHandlers>`. If the content of the view is defined using `UiBinder`, make sure that the type of the host object is a `PopopPanel`.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: host object is a ``PopupPanel`

File: site/src/main/markdown/gwtp_v2/core/presenters/popups.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    <g:PopupPanel autoHideOnHistoryEventsEnabled="true" autoHideEnabled="true" styleName="{style.popup}">
+    </g:PopupPanel>
+</ui:UiBinder>
+```
+
+## PopupPositioners
+
+GWTP provides a PopupPositioner to automatically position a popup.
+
+By default Popups use the CenterPopupPositioner which displays the Popup in the center of the screen. Popups will automatically be repositioned when the window is Resized.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: resized (remove capital R)?

File: site/src/main/markdown/gwtp_v2/core/presenters/view.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
+# The visual: View
+
+The second piece of the MVP architectural pattern is the View. Its purpose is to separate the application logic from the "display" logic. By implementing the UI this way, multiple views can be used to represent the same application (ie: different views for different countries, different platforms, etc). 
+
+The views in GWTP aren't much different from GWT widgets, except that they talk to a presenter. So the syntax remains pretty similar to GWT's. The preferred way of using complex views in GWT would be with UiBinder. A view can be defined by a declarative syntax in a `.ui.xml` file. Let's continue the previous example.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: previous example is ambiguous

File: site/src/main/markdown/gwtp_v2/core/presenters/view.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    
+    public void displayCurrentUserName(String username) {
+        this.username.setText(username);
+    }
+}
+```
+
+_The presenter will be ommited for now._
+
+The first things to see here, is the `extends` and `implements` on the class. In GWTP views have to extend an Implementation of the `View` interface. There exists multiple implementations available, but let's keep it simple with the regular `ViewImpl`. Also, a view should implement a Presenter's view contract. As mentioned in the Presenter section, a Presenter should declare the interface of its view, so this view implements its Presenter's view interface.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: The first things to notice here, are the

Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: In GWTP,

Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: remove the capital I on Implementation

Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: Maybe change There exists for There are. I think There exists is more common in mathematics

File: site/src/main/markdown/gwtp_v2/core/presenters/view.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+If the view uses the UiBinder (like here), it should declare its UiBinder's mapping. That's what is the `Binder` interface for. An instance of `Binder` has to be passed to the method `initWidget`. In GWTP, when it is possible, it is preferred to inject an instance by constructor.
```
Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: `An instance of Binder has to be passed to the method initWidget` : not really accurate, you pass the result of `binder.createAndBindUi(this)` to the initWidget method.

Comment (meriouma) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: Not sure `In GWTP, when it is possible, it is preferred to inject an instance by constructor` is clear

File: site/src/main/markdown/gwtp_v2/core/presenters/creating-places.md (hunk 4 above)
Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/creating-places.md]: Change the link to the doc on gwtproject.org

File: site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md (hunk 3 above)
Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: In Arcbees's drive under /Communication/PPT Presentation. I can't tell which one but I think it's from last GWT.Create.

File: site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+![Diagram](http://i.imgur.com/kwjjGuv.png)
+
+
+## Presenter's construction
+The first phase (which is omitted in this diagram) is the Presenter's construction. This happens when Gin calls the constructor of the Presenter. The only things that usually happen in a Presenter's contructor are:
+- Fields initialization
+- View's UI handlers initialization
+
+## onXXX()
+For all the following lifecycle methods, one should *always* call the corresponding `super.onXXX()` method. Otherwise, the lifecycle chain will be broken.
```
Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/lifecycle.md]: Well every onXX methods are no-ops. So it's only relevant if you have your personal abstracttion of PresenterWidget with shared onXX code.

File: site/src/main/markdown/gwtp_v2/core/presenters/popups.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+#Create a Popup Presenter
+
+##Presenter
+
+```java
+public class InfoPopupPresenterWidget extends PresenterWidget<InfoPopupPresenterWidget.MyView> {
+    /**
+     * {@link InfoPopupPresenterWidget}'s view.
+     */
+    public interface MyView extends PopupView {
```
Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: Remove public and the javadoc

Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: Maybe add some text or a more obvious pointer that the interface must extend PopupView

File: site/src/main/markdown/gwtp_v2/core/presenters/popups.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+    }
+}
+```
+
+##View
+The view extends `PopupViewImpl` or `PopupViewWithUiHandlers<PresentersUiHandlers>`. If the content of the view is defined using `UiBinder`, make sure that the type of the host object is a `PopupPanel`.
+
+```java
+public class InfoPopupView extends PopupViewImpl implements InfoPopupPresenterWidget.MyView {
+    public interface Binder extends UiBinder<PopupPanel, InfoPopupView> {
```
Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/popups.md]: remove public

File: site/src/main/markdown/gwtp_v2/core/presenters/view.md (hunk 12 above)
Comment (Chris-V) [inline site/src/main/markdown/gwtp_v2/core/presenters/view.md]: You could link to GWT's doc about UiBinder

**Comment by Chris-V** [comment]
Awesome job!

**Pull request merged**